flake.nix

```diff
@@ -0,0 +1,45 @@
+{
+  description =
+    "Tool to simplify secure shell connections over AWS Systems Manager (SSM).";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?ref=nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        awssh = pkgs.python310Packages.buildPythonPackage rec {
+          pname = "awssh";
+          version = "1.1.0";
+          doCheck = false;
+          propagatedBuildInputs = with pkgs.python310Packages; [
```

Comment:
Is this necessary? I would imagine that when you build the package it should pull in any dependencies specified in the package.

Reply:
No it doesn't work like that. This is necessary. 

>I would imagine that when you build the package it should pull in any dependencies specified in the package.

If we were using Poetry yes ;)

Reply:
Poetry has nothing to do with this either. If I install this package with `pip` it will pull in the dependencies listed in `setup.py`:
https://github.com/cisagov/awssh/blob/550a17d9d8dda7fcd4b523014a9344938a34a6eb/setup.py#L90-L95

Are you saying that Nix is unable to install a Python package using the existing Python dependency declaration functionality? Are these dependencies separate from the package's dependencies?

Reply:
Doesn't matter you still need to specify runtime dependencies. Also, this is not `pip`. The `poetry` comment  was to highlight the fact that you don't need to `pip` install anything. The toml declares what's needed and `poetry` will install those dependencies. It was just a joke.

>Are you saying that Nix is unable to install a Python package using the existing Python dependency declaration functionality? 
It can but it's messy, dirty and unnecessary. This is the better method. Not to mention the amount of time it will take to build all of that. This is faster. 

>Are these dependencies separate from the package's dependencies?
No they're not.

Reply:
Not to get too far afield here but per Poetry's documentation:

> Poetry, for its internal operations, uses the `pip` wheel embedded in the `virtualenv` package installed as a dependency in Poetry's runtime environment.